src/api.ts

```diff
@@ -122,29 +123,56 @@ export async function makeCoderSdk(baseUrl: string, token: string | undefined, s
 /**
  * Start or update a workspace and return the updated workspace.
  */
-export async function startWorkspaceIfStoppedOrFailed(restClient: Api, workspace: Workspace): Promise<Workspace> {
-  // If the workspace requires the latest active template version, we should attempt
-  // to update that here.
-  // TODO: If param set changes, what do we do??
-  const versionID = workspace.template_require_active_version
-    ? // Use the latest template version
-      workspace.template_active_version_id
-    : // Default to not updating the workspace if not required.
-      workspace.latest_build.template_version_id
-
+export async function startWorkspaceIfStoppedOrFailed(
+  restClient: Api,
+  binPath: string,
+  workspace: Workspace,
+  writeEmitter: vscode.EventEmitter<string>,
+): Promise<Workspace> {
   // Before we start a workspace, we make an initial request to check it's not already started
   const updatedWorkspace = await restClient.getWorkspace(workspace.id)
 
   if (!["stopped", "failed"].includes(updatedWorkspace.latest_build.status)) {
     return updatedWorkspace
   }
 
-  const latestBuild = await restClient.startWorkspace(updatedWorkspace.id, versionID)
+  return new Promise((resolve, reject) => {
+    const startProcess: ChildProcessWithoutNullStreams = spawn(binPath, [
+      "start",
+      "--yes",
+      workspace.owner_name + "/" + workspace.name,
```

Comment:
This will be the slightly tricky part I think.  For most folks this will also need `--global-config` so the cli can pick up the url and token.

`--global-config` expects `url` and `session` files in the directory to which it points, so it may be slightly tricky because the `vscodessh` command that the plugin has used up to this point takes separate url and file config flags (not sure why), and I guess as a consequence the token file (perhaps unintentionally) ended up with the slightly different name `session_token` (again not sure why the difference).

So all that to say, I think we will have to also make sure the `session` file exists, either by migrating from `session_token` to `session` or writing both, and then add the `--global-config` flag.

Reply:
Fixed by migrating `session_token` to `session` and adding the `--global-config` flag to `coder start`. Please let me know if this approach looks good.